mungegithub/mungers/triage-filer.go

```diff
@@ -0,0 +1,562 @@
+	Builds struct {
+		Cols struct {
+			Elapsed     []int    `json:"elapsed"`
+			Executor    []string `json:"executor"`
+			PR          []string `json:"pr"`
+			Result      []string `json:"result"`
+			Started     []int64  `json:"started"`
+			TestsFailed []int    `json:"tests_failed"`
+			TestsRun    []int    `json:"tests_run"`
+		} `json:"cols"`
+		JobsRaw  map[string]interface{} `json:"jobs"` // []int or map[string]int
+		Jobs     map[string]BuildToRowMap
+		JobPaths map[string]string `json:"job_paths"`
+	} `json:"builds"`
+	Clustered []struct {
+		ID    string `json:"id"`
+		Key   string `json:"key"`
+		Text  string `json:"text"`
+		Tests []struct {
+			Name string `json:"name"`
+			Jobs []struct {
+				Name   string `json:"name"`
+				Builds []int  `json:"builds"`
+			} `json:"jobs"`
+		} `json:"tests"`
+	} `json:"clustered"`
+}
+
+//BuildToRowMap is an interface that describes the buildnum to row index mapping used to retrieve data
+//about individual builds from the JSON file.
+//This is an interface because the JSON format describing failure clusters has 2 ways of recording the mapping info.
+type BuildToRowMap interface {
+	rowForBuild(buildnum int) (int, error)
+}
+
+//ContigRowMap is a BuildToRowMap implementation for when the buildnum to row index mapping describes
+//a contiguous set of rows via 3 ints.
+type ContigRowMap struct {
+	startRow, startBuild, count int
+}
+
+func (rowMap ContigRowMap) rowForBuild(buildnum int) (int, error) {
+	if buildnum < rowMap.startBuild || buildnum > rowMap.startBuild+rowMap.count-1 {
+		return 0, fmt.Errorf("Failed to find row in JSON for buildnumber: %d. Row mapping or buildnumber is invalid.", buildnum)
+	}
+	return buildnum - rowMap.startBuild + rowMap.startRow, nil
+}
+
+//DictRowMap is a BuildToRowMap implementation for when the buildnum to row index mapping is simply a dictionary.
+//The value type of this dictionary is interface instead of int so that we don't have to convert the original map.
+type DictRowMap map[string]interface{}
+
+func (rowMap DictRowMap) rowForBuild(buildnum int) (int, error) {
+	row, ok := rowMap[strconv.Itoa(buildnum)]
+	if !ok {
+		return 0, fmt.Errorf("Failed to find row in JSON for buildnumber: %d. Row mapping or buildnumber is invalid.", buildnum)
+	}
+	var irow float64
+	if irow, ok = row.(float64); !ok {
+		return 0, fmt.Errorf("Failed to find row in JSON for buildnumber: %d. Row mapping contains invalid type.", buildnum)
+	}
+	return int(irow), nil
+}
+
+//loadClusters generates a 'Cluster' struct for every cluster in the provided JSON that has
+//data that falls within the time window. Failures from before the time window
+//are ignored.
+func (filer *TriageFiler) loadClusters(jsonIn []byte) ([]*Cluster, error) {
+	data, err := parseTriageData(jsonIn)
+	if err != nil {
+		return nil, err
+	}
+
+	var clusters []*Cluster
+	//only consider failures that occured after this time
+	cutoffTime := time.Now().AddDate(0, 0, -filer.windowDays).Unix()
```

Comment:
It might be more appropriate to base this off the most recent build's time from the triage data instead of time.Now(), since the delay might be a few hours, shrinking the window size to 21 or 22 hours.